### ISSUE TITLE
Expose the client IP address to the server-side plugin code

### ISSUE THREAD

**Issue by evanderkoogh** [fab-spec/fab#216]
Most platforms have a way of exposing the IP address of the client. It would be great if FABs could specify a way to pass on that information to a plugin in a similar way that it already passes on the request, url, cookies and settings.

**Comment by geelen** [comment]
Yeah I considered this when doing the geolocation stuff on https://demo.fab.dev

I came across https://www.npmjs.com/package/request-ip, which I wanted to try to port across to FAB land but I didn't pursue it further. Sounds like all the information is already present on the request headers though?

In the end I decided it wasn't useful enough to make it a "standard" part of the FAB <-> Runtime interface. Seems like something that would be super easy to wrap up into a plugin?

```json5
plugins: {
  '@fab/plugin-request-ip': {},
  // ...
  './your-plugins-here': {},
}
```

The plugin itself would then be some variation on:

```ts
// @fab/plugin-request-ip/runtime.ts
import { FABRuntime, Priority } from '@fab/core'
// Import the whole library, or potentially inline the relevant logic
import requestIp from 'request-ip'

export default function({ Router }: FABRuntime) {
  Router.onAll(async ({ request, context }) => {
    context.request_ip = requestIp(request)
  }, Priority.FIRST)
}
```

Not sure if it'll be that simple, but that's what `context` is there for, a mutable storage for plugins to pass values between them.

Thoughts?

**Comment by evanderkoogh** [comment]
Hmm.. not a bad solution. But now we are relying on a third party library to work with all possibly hosting platforms.

And the hosting platform knows exactly what is needed to support just that particular platform. I doubt this would work with AWS Lambda@Edge for example. AFAIK the ip address is passed in the event, not in the headers.
